File: setup_dirs.py
import os

base = r'C:\Solera\WorkShop\IA WorkShop\gs-workshop-vaquita'
dirs = [
    'src/config',
    'src/db',
    'src/errors',
    'src/middleware',
    'src/modules/users',
    'src/modules/tandas',
    'src/test'
]

# Create directories
for d in dirs:
    full_path = os.path.join(base, d)
    os.makedirs(full_path, exist_ok=True)
    print(f"Created/verified: {full_path}")

print("\nDirectory structure:")
for root, dirs_list, files in os.walk(os.path.join(base, 'src')):
    print(root)

File: setup-dirs.js
const fs = require('fs');
const path = require('path');
const { execSync } = require('child_process');

const dirs = ['config','db','errors','models','repositories','services','validators','middleware','routes'];
const basePath = path.join(__dirname, 'src');

dirs.forEach(d => {
  const fullPath = path.join(basePath, d);
  fs.mkdirSync(fullPath, {recursive:true});
  console.log(`Created: ${fullPath}`);
});

console.log('All directories created successfully!');
console.log('\nInstalling jsonwebtoken and @types/jsonwebtoken...');
execSync('npm install jsonwebtoken @types/jsonwebtoken', { cwd: __dirname, stdio: 'inherit' });
console.log('npm install complete.');
